### PR TITLE
Documentation about using existing Grafana instance.

### PR DESCRIPTION
**Pull request by tbukic**
Signed-off-by: Bukic Tomislav, AVL <tomislav.bukic@avl.com>

Added some knowledge about using existing Grafana instances to the monitoring documentation as suggested [on Slack channel](https://ray-distributed.slack.com/archives/C03J169VAFP/p1668770358941089?thread_ts=1668502686.124139&cid=C03J169VAFP). 

Fixed version of PR https://github.com/ray-project/ray/pull/31633 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
This is an amazing PR! Thank you again @tbukic

File: doc/source/ray-observability/ray-metrics.rst [hunk 1]
```diff
@@ -87,6 +91,38 @@ You can then see the default dashboard by going to dashboards -> manage -> Ray -
 .. image:: images/graphs.png
     :align: center
 
+Using an existing Grafana instance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you want to use existing Grafana instance, before starting your Ray cluster you will need to setup environment variable `RAY_GRAFANA_HOST` with an URL of your Grafana. After starting Ray, you can find Grafana dashboard json at `/tmp/ray/session_latest/metrics/grafana/dashboards/default_grafana_dashboard.json`. `Import<https://grafana.com/docs/grafana/latest/dashboards/manage-dashboards/#import-a-dashboard>`_ this json to your Grafana.
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:97]: ``Import<https://grafana.com/docs/grafana/latest/dashboards/manage-dashboards/#import-a-dashboard>`_`

Can you tell me a bit about what this exactly means? the URL seems to be linked to the current documentation

File: doc/source/ray-observability/ray-metrics.rst [hunk 2]
```diff
@@ -87,6 +91,38 @@ You can then see the default dashboard by going to dashboards -> manage -> Ray -
...
+
+If Grafana reports that datasource is not found, you can `add a datasource variable <https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/?pg=graf&plcmt=data-sources-prometheus-btn-1#add-a-data-source-variable>`_ and change all values of `datasource` key in the dashboard json to the name of the variable. For example, if the variable name is `data_source`, all `"datasource"` mappings should be:
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:99]: "key in the dashboard json"

Is this `default_grafana_dashboard.json`? Can you change this part to

"key in the `default_grafana_dashboard.json`"?

File: doc/source/ray-observability/ray-metrics.rst [hunk 3]
```diff
@@ -87,6 +91,38 @@ You can then see the default dashboard by going to dashboards -> manage -> Ray -
...
+  cookie_samesite = none
+
+If Grafana is exposed via nginx ingress on Kubernetes cluster, the following line should be present in the Grafana ingress annotation:
+
+.. code-block:: yaml
+
+  nginx.ingress.kubernetes.io/configuration-snippet: |
+      add_header X-Frame-Options SAMEORIGIN always;
+
+When both Grafana and Ray cluster are on the same Kubernetes cluster, it is important to set `RAY_GRAFANA_HOST` to the external URL of the Grafana ingress. For successful embedding, `RAY_GRAFANA_HOST` needs to be accessible to both Ray cluster backend and Ray dashboard frontend: backend does the healthchecks and frontend embeds the dashboard.
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:124]: Can you also add the difference between backend / frontend? E.g., 

backend (ray cluster) and frontend (machines that accesses the dashboard URL)

**Review by rkooo567 — APPROVED** [review]

File: doc/source/ray-observability/ray-metrics.rst (hunk 1 above)
Comment (tbukic) [reply doc/source/ray-observability/ray-metrics.rst:97]: Json at `/tmp/ray/session_latest/metrics/grafana/dashboards/default_grafana_dashboard.json` is encoding Grafana dashboard. When using existing Grafana, user needs to manually import that json. It is pretty straightforward, but Grafana documentation is showing how it is done, step-by-step. I am going to rephrase it.

File: doc/source/ray-observability/ray-metrics.rst (hunk 3 above)
Comment (tbukic) [reply doc/source/ray-observability/ray-metrics.rst:124]: Thank you. I've extended on this as well.

**Comment by rkooo567** [comment]
I need an approval from the doc owners! I have asked for the approval, so it will be approved soon!

**Review by maxpumperla — APPROVED** [review]
Thank you @tbukic !

**Pull request merged**